jobs/schedulers.go

```diff
@@ -232,13 +232,26 @@ func (schedulers *Schedulers) scheduleJob(cfg *model.Config, scheduler model.Sch
 
 func (schedulers *Schedulers) handleConfigChange(oldConfig, newConfig *model.Config) {
 	mlog.Debug("Schedulers received config change.")
-	schedulers.configChanged <- newConfig
+	select {
+	case schedulers.configChanged <- newConfig:
+	case <-schedulers.stop:
+	}
 }
 
-func (schedulers *Schedulers) HandleClusterLeaderChange(isLeader bool) {
+func (schedulers *Schedulers) handleClusterLeaderChange(isLeader bool) {
 	select {
 	case schedulers.clusterLeaderChanged <- isLeader:
 	default:
-		mlog.Debug("Did not send cluster leader change message to schedulers as no schedulers listening to notification channel.")
+		mlog.Debug("Sending cluster leader change message to schedulers failed.")
```

Comment:
We get there if the channel is full. That means that an event already got there (the chan is buffered of length 1) but the goroutine in `Start()` has not yet handled it either because it was not running or doing something else.

By how we moved things around now it would be a rather rare event but still possible so I wouldn't regard it as a warning really. Before the changes we would lose the message completely so it was a problem but now we remove the existing one and enqueue the latest so eventually the latest message will get there (unless the server shuts down before that of course).